front/src/components/oauth2/actions.js

```diff
@@ -0,0 +1,109 @@
+import { RequestStatus } from '../../utils/consts';
+import { OAUTH2 } from '../../../../server/utils/constants';
+
+const actions = store => ({
+  updateIntegrationName(state, e) {
+    store.setState({
+      integrationName: e
+    });
+  },
+  updateClientId(state, e) {
+    store.setState({
+      clientId: e.target.value
+    });
+  },
+  updateSecret(state, e) {
+    store.setState({
+      secret: e.target.value
+    });
+  },
+  async getCurrentConfig(state) {
+    store.setState({
+      oauth2GetStatus: RequestStatus.Getting
+    });
+
+    const returnServiceId = (await state.httpClient.get(`/api/v1/service/${state.integrationName}`)).id;
+
+    const returnGetConfig = await state.httpClient.get('/api/v1/service/oauth2/client', {
+      service_id: returnServiceId
+    });
+    if (returnGetConfig) {
+      store.setState({
+        clientIdInDb: returnGetConfig.client_id,
+        oauth2ErrorMsg: null,
+        oauth2GetStatus: RequestStatus.Success
+      });
+    } else {
+      store.setState({
+        oauth2ErrorMsg: null,
+        oauth2GetStatus: RequestStatus.Success
+      });
+    }
+  },
+  async startConnect(state) {
+    store.setState({
+      oauth2GetStatus: RequestStatus.Getting
+    });
+    try {
+      if (!state.secret || !state.clientId) {
+        store.setState({
+          oauth2GetStatus: RequestStatus.Error,
+          oauth2ErrorMsg: 'errorEmptyConfig'
+        });
+        return;
+      }
+
+      const serviceId = (await state.httpClient.get(`/api/v1/service/${state.integrationName}`)).id;
```

Comment:
I don't understand, in this case I use common service controller of the integration I develop.

Reply:
There is common `actions/integration.js` which does exactly the same, but in the "integration dedicated action file".
You may need to re-use it, instead of duplicating on this and other sides.

Reply:
ok I fix it